lib/reversetunnel/seek/seek.go

```diff
@@ -248,6 +251,10 @@ func (p *proxyGroup) run(ctx context.Context) {
 			if stat.ShouldSeek() {
 				p.notifyShouldSeek()
 			}
+			ticks++
+			if ticks%logInterval == 0 {
+				log.Debugf("SeekStates(states=%+v,id=%s)", p.GetStates(), p.id)
```

Comment:
here you can take advantage of the structured logging, also can you port this to a prometheus gauge instead to report the different states?

Reply:
Not very familiar with prometheus.  As far as I can tell, it only deals with scalar values.  The value being logged here is effectively a `map[string]string`.  Is there a mechanism I'm missing that is appropriate for this?

Side Note; I can definitely plug the more frequently calculated *summary* values into prometheus.  It won't indicate *which* proxies are in which state, but it will give a nice overview of the *number* of proxies in each state.  Think that would be sufficient?

Reply:
Yes, I think this will be OK

Reply:
So I added prometheus gauge for the *connected* proxies, but it turns out that complete statistics require rework that I'd prefer to tackle sometime *after* `4.1` release.  Currently, there isn't a good "single source of truth" for most of the process-level stats, and its surprisingly easy to get some of the values wrong.